ml_dtypes/_src/custom_float.h

```diff
@@ -870,35 +870,9 @@ bool RegisterFloatDtype(PyObject* numpy, bool* already_registered = nullptr) {
   if (already_registered != nullptr) {
     *already_registered = false;
   }
-  // If another module (presumably either TF or JAX) has registered a bfloat16
-  // type, use it. We don't want two bfloat16 types if we can avoid it since it
-  // leads to confusion if we have two different types with the same name. This
-  // assumes that the other module has a sufficiently complete bfloat16
-  // implementation. The only known NumPy bfloat16 extension at the time of
-  // writing is this one (distributed in TF and JAX).
-  // TODO(phawkins): distribute the bfloat16 extension as its own pip package,
-  // so we can unambiguously refer to a single canonical definition of bfloat16.
-  int typenum =
-      PyArray_TypeNumFromName(const_cast<char*>(TypeDescriptor<T>::kTypeName));
-  if (typenum != NPY_NOTYPE) {
-    PyArray_Descr* descr = PyArray_DescrFromType(typenum);
-    // The test for an argmax function here is to verify that the
-    // bfloat16 implementation is sufficiently new, and, say, not from
-    // an older version of TF or JAX.
-    if (descr && descr->f && descr->f->argmax) {
-      TypeDescriptor<T>::npy_type = typenum;
-      TypeDescriptor<T>::type_ptr = reinterpret_cast<PyObject*>(descr->typeobj);
-      if (already_registered != nullptr) {
-        *already_registered = true;
-      }
-      return true;
-    }
-  }
-
   // It's important that we heap-allocate our type. This is because tp_name
```

Comment:
I'll note that even this "heap-allocate" bit probably isn't true, any more. We did that so the names agreed, but we're not looking for the names any more and not trying to interoperate with anyone who is.

TODO to remove?

Reply:
Makes sense – beyond that we probably don't need any of the machinery related to `already_registered`. I'll add a TODO to simplify this. I'd like to get this in as-is, though, because it will unblock testing JAX against numpy's nightly release.